### PR TITLE
candidate search, mapping, and CC Sunlight API calls

### PR DESCRIPTION
**Pull request by kavunshiva**
- [x] add method to order candidates by attribute
- [x] add address formatting
- [x] add method to call `CC Sunlight API`
- [x] add method to map numbers to column letter names

### CODE REVIEW

File: query_sheet.rb [hunk 1]
```diff
@@ -186,5 +208,126 @@ def add_errors_to_message(
...
+    if obj[candidate[attribute]]
+      obj[candidate[attribute]] << candidate
+    else
+      obj[candidate[attribute]] = one_to_many ? [candidate] : candidate
+    end
+    obj
+  end
+end
+
+def find_diffs(imported_candidates)
```
Comment (thomasturnbull) [inline query_sheet.rb]: May as well remove this, my branch has something very similar.

File: query_sheet.rb [hunk 2]
```diff
@@ -186,5 +208,126 @@ def add_errors_to_message(
...
+    candidate['primary_city'] &&
+    candidate['primary_city'].length > 0 &&
+    candidate['primary_city'].match(/brooklyn|new york/i) &&
+    candidate['primary_state'] &&
+    candidate['primary_state'].length > 0 &&
+    candidate['primary_state'].match(/ny/i) &&
+    candidate['primary_zip'] &&
+    candidate['primary_zip'].length > 0 &&
+    candidate['primary_zip'].length == 5
+      return nil
```
Comment (thomasturnbull) [inline query_sheet.rb]: Perhaps add a TODO to move into separate method, and do something if invalid.

File: query_sheet.rb [hunk 3]
```diff
@@ -186,5 +208,126 @@ def add_errors_to_message(
...
+  candidate['state_lower_district'] == candidate['ad']
+end
+
+def ad_match?(candidiate, new_candidate)
+  candidate['ad'] == new_candidate['ad']
+end
+
+# spreadsheets_columns_valid?
+
+# find_diffs(
```
Comment (thomasturnbull) [inline query_sheet.rb]: Remove?

File: query_sheet.rb [hunk 4]
```diff
@@ -186,5 +208,126 @@ def add_errors_to_message(
...
+#   read_sheet(
+#     '1Jl_Gr-WcRstFhHNHGOVin9IAxfuCaXhNDnOAqOfHY8s',
+#     'nationbuilder-people-export-2019-07-09-2131',
+#   )
+# )
 validate_spreadsheets_columns
-# puts read_sheet(ARGV[0], ARGV[1])
+
+binding.pry
```
Comment (thomasturnbull) [inline query_sheet.rb]: Remove?

File: query_sheet.rb (hunk 1 above)
Comment (kavunshiva) [reply query_sheet.rb]: #deprecation

File: query_sheet.rb (hunk 2 above)
Comment (kavunshiva) [reply query_sheet.rb]: not a bad idea

File: query_sheet.rb (hunk 3 above)
Comment (kavunshiva) [reply query_sheet.rb]: yeah, might as well: it's kind of useless

File: query_sheet.rb (hunk 4 above)
Comment (kavunshiva) [reply query_sheet.rb]: yessir

**Comment by kavunshiva** [comment]
@thomasturnbull made the requested changes; have another looksie when you're ready

**Pull request merged**